tests/ingestion.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import json
+import pystac
+from pystac import STACValidationError
+import os
+import boto3
+from dotenv import load_dotenv
+
+load_dotenv()
+
+class StacIngestion:
+
+    """Class representing various test operations"""
+
+    def __init__(self):
+        self.base_url = os.getenv("BASE_URL")
+        self.stac_url = os.getenv("STAC_URL")
+        self.collections_endpoint = os.getenv("COLLECTIONS_ENDPOINT")
+        self.items_endpoint = os.getenv("ITEMS_ENDPOINT")
+        self.current_file_path = os.path.dirname(os.path.abspath(__file__))
+
+    def validate_collection(self, collection):
+        try:
+            pystac.validation.validate_dict(collection)
+        except STACValidationError:
+            raise STACValidationError("Validation failed for the collection")
+
+    def validate_item(self, item):
+        try:
+            pystac.validation.validate_dict(item)
+        except STACValidationError:
+            raise STACValidationError("Validation failed for the item")
+
+    def get_authentication_token(self):
+        #Authentication - Get secret value
+        stage = os.getenv("STAGE")
+        stac_register_service_id = os.getenv("STAC_REGISTER_SERVICE_ID")
+        stack_name = f"MAAP-STAC-auth-{stage}"
+
+        # session = boto3.session.Session()
+        client = boto3.client("secretsmanager", region_name=os.getenv("REGION_NAME"))
+        secret_id = f"{stack_name}/{stac_register_service_id}"
+
+        try:
+            res_secret = client.get_secret_value(SecretId=secret_id)
+        except client.exceptions.ResourceNotFoundException:
+            raise Exception(
+                f"Unable to find a secret for '{secret_id}'. "
+                "\n\nHint: Check your stage and service id. Also, verify that the correct "
+                "AWS_PROFILE is set on your environment."
+            )
+
+        #Authentication - Get TOKEN
+        secret = json.loads(res_secret["SecretString"])
+        client_secret = secret["client_secret"]
+        client_id = secret["client_id"]
+        cognito_domain = os.getenv("COGNITO_DOMAIN")
+        scope = os.getenv("SCOPE")
+
+        res_token = requests.post(
+                    f"{cognito_domain}/oauth2/token",
+                    headers={
+                        "Content-Type":"application/x-www-form-urlencoded",
+                    },
+                    auth=(client_id, client_secret),
+                    data={
+                        "grant_type":"client_credentials",
+                        # A space-separated list of scopes to request for the generated access token.
+                        "scope": scope,
+                    },
+                )
+
+        authentication_token = res_token.json()["access_token"]
```

Comment:
Can we rename the authentication_token variable differently than the authentication_token function?